File: src/main/kotlin-graphql/io/github/graphql/kotlin/model/CodeOfConductTO.kt
package io.github.graphql.kotlin.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.Objects
import java.util.StringJoiner

@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-07T19:48:13+0800"
)
open class CodeOfConductTO(
    val body: String?,
    override
    val id: String,
    val key: String,
    val name: String,
    val resourcePath: String?,
    val url: String?
) : NodeTO {

    companion object {
        fun builder(): Builder = Builder()
    }

    // In the future, it maybe change.
    override fun toString(): String {
        val joiner = StringJoiner(", ", "{ ", " }")
        if (body != null) {
            joiner.add("body: " + GraphQLRequestSerializer.getEntry(body))
        }
        joiner.add("id: " + GraphQLRequestSerializer.getEntry(id))
        joiner.add("key: " + GraphQLRequestSerializer.getEntry(key))
        joiner.add("name: " + GraphQLRequestSerializer.getEntry(name))
        if (resourcePath != null) {
            joiner.add("resourcePath: " + GraphQLRequestSerializer.getEntry(resourcePath))
        }
        if (url != null) {
            joiner.add("url: " + GraphQLRequestSerializer.getEntry(url))
        }
        return joiner.toString()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as CodeOfConductTO
        return Objects.equals(body, that.body)
                && Objects.equals(id, that.id)
                && Objects.equals(key, that.key)
                && Objects.equals(name, that.name)
                && Objects.equals(resourcePath, that.resourcePath)
                && Objects.equals(url, that.url)

    }

    override fun hashCode(): Int = {
        return Objects.hash(body, id, key, name, resourcePath, url)
    }

    class Builder {

        private var body: String? = null
        private lateinit var id: String
        private lateinit var key: String
        private lateinit var name: String
        private var resourcePath: String? = null
        private var url: String? = null

        fun setBody(body: String?): Builder {
            this.body = body
            return this
        }

        fun setId(id: String): Builder {
            this.id = id
            return this
        }

        fun setKey(key: String): Builder {
            this.key = key
            return this
        }

        fun setName(name: String): Builder {
            this.name = name
            return this
        }

        fun setResourcePath(resourcePath: String?): Builder {
            this.resourcePath = resourcePath
            return this
        }

        fun setUrl(url: String?): Builder {
            this.url = url
            return this
        }

        fun build(): CodeOfConductTO {
            return CodeOfConductTO(body, id, key, name, resourcePath, url)
        }
    }
}

File: src/main/scala-graphql/io/github/graphql/model/CodeOfConductTO.scala
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import scala.collection.JavaConverters._

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-06T19:48:00+0800"
)
class CodeOfConductTO(
    body: String,
    @javax.validation.constraints.NotNull
    override val id: String,
    @javax.validation.constraints.NotNull
    key: String,
    @javax.validation.constraints.NotNull
    name: String,
    resourcePath: String,
    url: String
) extends NodeTO {

    override def toString(): String = {
        Seq(
            if (body != null) "body: " + GraphQLRequestSerializer.getEntry(body) else "",
            if (id != null) "id: " + GraphQLRequestSerializer.getEntry(id) else "",
            if (key != null) "key: " + GraphQLRequestSerializer.getEntry(key) else "",
            if (name != null) "name: " + GraphQLRequestSerializer.getEntry(name) else "",
            if (resourcePath != null) "resourcePath: " + GraphQLRequestSerializer.getEntry(resourcePath) else "",
            if (url != null) "url: " + GraphQLRequestSerializer.getEntry(url) else ""
        ).filter(_ != "").mkString("{", ",", "}")
    }
}

object CodeOfConductTO {

    def builder(): CodeOfConductTO.Builder = new Builder()

    class Builder {

        private var body: String = _
        private var id: String = _
        private var key: String = _
        private var name: String = _
        private var resourcePath: String = _
        private var url: String = _

        def setBody(body: String): Builder = {
            this.body = body
            this
        }

        def setId(id: String): Builder = {
            this.id = id
            this
        }

        def setKey(key: String): Builder = {
            this.key = key
            this
        }

        def setName(name: String): Builder = {
            this.name = name
            this
        }

        def setResourcePath(resourcePath: String): Builder = {
            this.resourcePath = resourcePath
            this
        }

        def setUrl(url: String): Builder = {
            this.url = url
            this
        }

        def build(): CodeOfConductTO = new CodeOfConductTO(body, id, key, name, resourcePath, url)

    }
}
